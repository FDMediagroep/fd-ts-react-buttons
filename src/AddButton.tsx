import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export type ButtonTypes = 'default' | 'persoonlijk';

export interface Props {
    buttonStyle: ButtonTypes;
    className?: string;
    onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

export default class AddButton extends PureComponent<any, any> {
    render() {
        return (
            <>
                <GlobalStyle/>
                <button className={`fd-add-button${this.props.className ? ` ${this.props.className}` : ''}${this.props.buttonStyle ? ` ${this.props.buttonStyle}` : ''}`} onClick={this.props.onClick}>{this.props.children}</button>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-add-button {
    outline: none; /* Sorry, visually impaired, I tried */
    position: relative;
    border-radius: 2px;
    white-space: nowrap;
    padding: 5px 10px;
    border: 1px solid #677381;
    color: #ffeadb;
    background-color: #677381;
    &:hover, &:active, &:focus {
        background-color: #2e3843;
        border-color: #2e3843;
    }
    cursor: pointer;
    transition: background-color .2s;
    font-family: 'ProximaNovaRegular', sans-serif;
    font-size: 14px;

    &.persoonlijk {
        color: #ffffff;
    }
}
`;
export {GlobalStyle as AddButtonStyle};
