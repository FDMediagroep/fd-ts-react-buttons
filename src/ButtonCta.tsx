import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export default class ButtonCallToAction extends PureComponent<any, any> {
    render() {
        return (
            <>
                <GlobalStyles/>
                <button {...this.props} className={`fd-button-cta${this.props.className ? ` ${this.props.className}` : ''}`}>{this.props.children}</button>
            </>
        );
    }
}

const GlobalStyles = createGlobalStyle`
.fd-button-cta {
    color: #FFFFFF;
    font-family: ProximaNovaBold;
    font-size: 1rem;
    border: none;
    text-align: center;
    background-color: #ff8529;
    :hover {
        background-color: #DF680E;
    }
    box-shadow: rgba(0, 0, 0, 0.2) 0px -2px 0px 0px inset;
    border-radius: 3px;
    padding: .5rem 1rem;
    line-height: 1.2;
    cursor: pointer;
    &.m {
        font-size: 18px;
        padding: .75rem 1.5rem;
    }
    &.l {
        font-size: 20px;
        padding: .75rem 2rem;
    }
}
`;

export const ButtonCallToActionStyles = GlobalStyles;
