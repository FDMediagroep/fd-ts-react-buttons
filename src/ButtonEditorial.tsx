import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export default class ButtonEditorial extends PureComponent<any, any> {
    render() {
        return (
            <>
                <GlobalStyles/>
                <button {...this.props} className={`fd-button${this.props.className ? ` ${this.props.className}` : ''}`}>{this.props.children}</button>
            </>
        );
    }
}

const GlobalStyles = createGlobalStyle`
.fd-button {
    color: #FFFFFF;
    font-family: ProximaNovaBold;
    font-size: 1rem;
    border: none;
    text-align: center;
    background-color: #677381;
    :hover {
        background-color: #2e3843;
    }
    box-shadow: rgba(0, 0, 0, 0.2) 0px -2px 0px 0px inset;
    border-radius: 3px;
    padding: .5rem 1rem;
    line-height: 1.2;
    cursor: pointer;
    &.m {
        font-size: 18px;
        padding: .75rem 1.5rem;
    }
    &.l {
        font-size: 20px;
        padding: .75rem 2rem;
    }
}
`;

export const ButtonEditorialStyles = GlobalStyles;
