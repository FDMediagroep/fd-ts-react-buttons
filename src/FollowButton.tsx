import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export type ButtonTypes = 'default' | 'persoonlijk';

export interface Props {
    buttonStyle: ButtonTypes;
    className?: string;
    followButtonText?: string;
    followLink?: string;
    unFollowButtonText?: string;
    unFollowLink?: string;
    onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
    selected?: boolean;
    tag: string;
}

export default class FollowButton extends PureComponent<Props, any> {
    render() {
        return (
            <>
                <GlobalStyle/>
                <button className={`fd-follow-button${this.props.className ? ` ${this.props.className}` : ''}${this.props.buttonStyle ? ` ${this.props.buttonStyle}` : ''}`} onClick={this.props.onClick} data-tag={this.props.tag} data-selected={this.props.selected} data-addurl={this.props.followLink ? this.props.followLink : '/add-interest'} data-deleteurl={this.props.unFollowLink ? this.props.unFollowLink : '/delete-interest'}>
                    <span className="cross"/> <span className="button-text default-text">{this.props.followButtonText ? this.props.followButtonText : 'Volg'}</span><span className="button-text active-text">{this.props.unFollowButtonText ? this.props.unFollowButtonText : 'Ontvolg'}</span>
                </button>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-follow-button {
    position: relative;
    outline: none;
    border-radius: 2px;
    white-space: nowrap;
    padding: 5px 10px;
    border: 1px solid #49a4a2;
    color: #49a4a2;
    background-color: transparent;
    cursor: pointer;
    transition: background-color .2s;
    font-family: 'ProximaNovaRegular', sans-serif;
    font-size: 14px;

    &:hover {
        background-color: #49a4a2;
        color: #ffeadb;
    }
    .cross {
        &:before {
            content: '+';
            display: block;
            width: 10px;
            height: 10px;
            line-height: 10px;
        }
        display: inline-block;
        transform: rotate(0deg);
        transition: transform .5s;
        transform-origin: center center;
        font-size: 20px;
        position: relative;
        top: 2px;
    }

    .button-text {
        margin-left: 5px;
    }

    .active-text {
        display: none;
    }

    &[data-selected="true"] {
        background-color: #49a4a2;
        color: #ffeadb;
        .cross {
            transform: rotate(45deg);
        }

        .default-text {
            display: none;
        }
        .default-icon,
        .active-text {
            display: inline;
        }
    }

    &.persoonlijk {
        .cross,
        .button-text {
            color: #000000;
        }

        &:hover {
            .cross,
            .button-text {
                color: #ffffff;
            }
        }

        &[data-selected="true"] {
            .button-text, .cross {
                color: #ffffff;
            }
        }
    }
}
`;
export {GlobalStyle as FollowButtonStyle};
