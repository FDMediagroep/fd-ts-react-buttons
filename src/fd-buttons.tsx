import React from "react";
import styled from 'styled-components';

export interface Props {
    type?: 'button' | 'submit';
    layout?: 'primary' | 'secondary';
    children?: React.ReactNode;
    className?: string;
    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export default class Button extends React.PureComponent<Props, any> {
    render() {
        const classes: string[] = [];
        if (this.props.layout) {
            classes.push(this.props.layout);
        }
        if (this.props.className) {
            classes.push(this.props.className);
        }
        return (
            <StyledButton
                onClick={this.props.onClick}
                type={this.props.type ? this.props.type : 'button'}
                className={classes.join(' ')}
            >
                {this.props.children}
            </StyledButton>
        );
    }
}

const StyledButton = styled('button')`
    background-color: orange;
`;
