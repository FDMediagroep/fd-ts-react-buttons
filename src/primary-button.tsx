import React from "react";

export interface Props {
    type?: string;
}

export default class PrimaryButton extends React.PureComponent<Props, any> {
    render() {
        return (
            <button
                type={this.props.type}
            ></button>
        );
    }
}
