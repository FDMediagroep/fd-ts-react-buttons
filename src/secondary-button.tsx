import React from "react";

export interface Props {
    type?: string;
}

export default class SecondaryButton extends React.PureComponent<Props, any> {
    render() {
        return (
            <button
                type={this.props.type}
            ></button>
        );
    }
}
