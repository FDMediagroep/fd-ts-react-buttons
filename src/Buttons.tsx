import AddButton, {AddButtonStyle} from './AddButton';
import ButtonCallToAction, {ButtonCallToActionStyles} from './ButtonCta';
import ButtonEditorial, { ButtonEditorialStyles } from './ButtonEditorial';
import FollowButton, { FollowButtonStyle } from './FollowButton';

export {
    AddButton, AddButtonStyle,
    ButtonCallToAction, ButtonCallToActionStyles,
    ButtonEditorial, ButtonEditorialStyles,
    FollowButton, FollowButtonStyle
};
