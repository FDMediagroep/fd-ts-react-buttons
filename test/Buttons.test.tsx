import TestRenderer from 'react-test-renderer';
import {ButtonCallToAction, ButtonEditorial} from '../src/Buttons';
import React from 'react';

describe('Button', () => {
    test('renders call-to-actions button correctly', () => {
        let button = TestRenderer.create(<ButtonCallToAction/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<ButtonCallToAction className="m"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<ButtonCallToAction className="l"/>);
        expect(button.toJSON()).toMatchSnapshot();
    });

    test('renders editorial buttons correctly', () => {
        let button = TestRenderer.create(<ButtonEditorial/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<ButtonEditorial className="m"/>);
        expect(button.toJSON()).toMatchSnapshot();
        button = TestRenderer.create(<ButtonEditorial className="l"/>);
        expect(button.toJSON()).toMatchSnapshot();
    });
});