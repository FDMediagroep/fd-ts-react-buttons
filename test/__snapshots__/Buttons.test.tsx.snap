// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders call-to-actions button correctly 1`] = `
<button
  className="fd-button-cta"
/>
`;

exports[`Button renders call-to-actions button correctly 2`] = `
<button
  className="fd-button-cta m"
/>
`;

exports[`Button renders call-to-actions button correctly 3`] = `
<button
  className="fd-button-cta l"
/>
`;

exports[`Button renders editorial buttons correctly 1`] = `
<button
  className="fd-button"
/>
`;

exports[`Button renders editorial buttons correctly 2`] = `
<button
  className="fd-button m"
/>
`;

exports[`Button renders editorial buttons correctly 3`] = `
<button
  className="fd-button l"
/>
`;
